python/lsst/meas/base/compensatedGaussian/_compensatedGaussian.py

```diff
@@ -119,44 +105,68 @@ def __init__(
             mask_str = f"{base_key}_mask_bits"
             mask_key = schema.addField(mask_str, type=np.int32, doc="Mask bits set within aperture.")
 
-            self.width_keys[width] = (flux_key, err_key, mask_key)
+            # failure flags
+            failure_flag = flagDefs.add(f"{width}_flag", "Compensated Gaussian measurement failed")
+            oob_flag = flagDefs.add(f"{width}_flag_bounds", "Compensated Gaussian out-of-bounds")
+
+            self.width_keys[width] = (flux_key, err_key, mask_key, failure_flag, oob_flag)
             self._rads[width] = math.ceil(sps.norm.ppf((0.995,), scale=width * config.t)[0])
 
+        self.flagHandler = FlagHandler.addFields(schema, name, flagDefs)
         self._max_rad = max(self._rads)
 
     def fail(self, measRecord, error=None):
-        if isinstance(error, OutOfBoundsError):
-            measRecord.set(self.ooBoundsKey, True)
-        measRecord.set(self.fatalFailKey, True)
+        """Record failure
+
+        See also
+        --------
+        lsst.meas.base.SingleFramePlugin.fail
+        """
+        if error is None:
+            self.flagHandler.handleFailure(measRecord)
+        else:
+            self.flagHandler.handleFailure(measRecord, error.cpp)
 
     def measure(self, measRecord, exposure):
         center = measRecord.getCentroid()
         bbox = exposure.getBBox()
 
-        if Point2I(center) not in exposure.getBBox().erodedBy(self._max_rad):
-            raise OutOfBoundsError("Not all the kernels for this source fit inside the exposure.")
-
         y = center.getY() - bbox.beginY
         x = center.getX() - bbox.beginX
 
         y_floor = math.floor(y)
         x_floor = math.floor(x)
 
-        for width, (flux_key, err_key, mask_key) in self.width_keys.items():
+        for width, (flux_key, err_key, mask_key, failure_flag, oob_flag) in self.width_keys.items():
             rad = self._rads[width]
+
+            if Point2I(center) not in exposure.getBBox().erodedBy(rad):
+                self.flagHandler.setValue(measRecord, failure_flag.number, True)
+                self.flagHandler.setValue(measRecord, oob_flag.number, True)
+                continue
+
             y_slice = slice(y_floor - rad, y_floor + rad + 1, 1)
             x_slice = slice(x_floor - rad, x_floor + rad + 1, 1)
             y_mean = y - y_floor + rad
             x_mean = x - x_floor + rad
 
+            sub_im = exposure.image.array[y_slice, x_slice]
+            sub_var = exposure.variance.array[y_slice, x_slice]
```

Comment:
You could technically move setting `sub_var` to after the size checking (but readability may win out to leaving here...up to you).